src/Google.Api.Gax/ApiCallRetryExtensions.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Google.Protobuf;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax
+{
+    public static class ApiCallRetryExtensions
+    {
+        public static ApiCall<TRequest, TResponse> WithRetry<TRequest, TResponse>(
+            this ApiCall<TRequest, TResponse> originalCall,
+            RetrySettings retrySettings,
+            IClock clock,
+            IScheduler scheduler)
+            where TRequest : class, IMessage<TRequest>
+            where TResponse : class, IMessage<TResponse>
+        {
+            if (retrySettings == null)
```

Comment:
It'll make the code-gen'ed code slightly nicer if it transparently handles nulls.
E.g. It'll be able to do this:
_createTopic = _clientHelper.BuildApiCall(...).WithRetry(effectiveSettings.CreateTopicRetry)
And if effectiveSettings.CreateTopicRetry has been set to null (by user code) then it all "just works" correctly.